File: FindMyHome.Frontend/src/app/models/listing/listing-edit.model.ts
export class ListingEdit{
    public listingId : number;
    public listingTypeId : number;
    public listingMarketingTypeId: number;
    public countyId : number;
    public cityId : number;
    public title : string;
    public description : string;
    public location: string;
    public price : number;
    public pictures : string[] | undefined;
    public latitude: number | undefined;
    public longitude: number | undefined;

    public constructor(carId: number,
                       listingType: number,
                       listingMarketingType: number,
                       countyId: number, 
                       cityId : number, 
                       title : string, 
                       description : string,
                       location: string,
                       price : number, 
                       photo : string[] | undefined,
                       latitude: number | undefined,
                       longitude: number | undefined) {
        this.listingId = carId;                   
        this.listingTypeId = listingType;
        this.listingMarketingTypeId = listingMarketingType;
        this.countyId = countyId;
        this.cityId = cityId;
        this.title = title;
        this.description = description;
        this.location = location;
        this.price = price;
        this.pictures = photo;
        this.latitude = latitude;
        this.longitude = longitude;
    }
}

File: FindMyHome.Domain/DTOs/Listing/ListingEditModel.cs
namespace FindMyHome.Domain.DTOs.Listing;

public class ListingEditModel
{
    public int ListingId { get; set; }
    public string Title { get; set; }
    public string Description { get; set; }
    public string Location { get; set; }
    public int ListingTypeId { get; set; }
    public int ListingMarketingTypeId { get; set; }
    public int CountyId { get; set; }
    public int CityId { get; set; }
    public long Price { get; set; }
    public long? Latitude { get; set; }
    public long? Longitude { get; set; }
    public bool IsClosed { get; set; }
    public string? Picture { get; set; }
    public int UserId { get; set; }
}
